### PR TITLE
test violation locations

### PR DESCRIPTION
**Pull request by jpsim**
We've had several issues in the past due to violations being reported at the wrong location. This PR introduces a *violation marker* (`↓`) in rule description trigger strings that gets stripped out when linting the string, but is used to identify where violations should occur.

With this approach, we should be able to insert arbitrary multibyte characters while confirming that the violation is still triggered at the appropriate location. I'll make a new PR doing just that once this one has been reviewed. /cc @norio-nomura

### CODE REVIEW

File: Source/SwiftLintFrameworkTests/TestHelpers.swift [hunk 1]
```diff
@@ -65,6 +70,18 @@ extension XCTestCase {
 
         // Triggering examples violate
         XCTAssertEqual(triggers.flatMap({ violations($0, ruleDescription) }).count, triggers.count)
+        let triggersWithMarkers = triggers.filter { $0.containsString(violationMarker) }
+
+        // Triggering examples with violation markers violate at the marker's location
+        for trigger in triggersWithMarkers {
+            let firstViolation = violations(trigger, ruleDescription).first!
```
Comment (keith) [inline Source/SwiftLintFrameworkTests/TestHelpers.swift]: Maybe we should combine this and the `triggers.flatMap` above to only calculate the violations once?

**Comment by keith** [comment]
This is great! Might be worth combining those from my comment and :ship:

**Comment by norio-nomura** [comment]
:+1:

File: Source/SwiftLintFrameworkTests/TestHelpers.swift (hunk 1 above)
Comment (jpsim) [inline Source/SwiftLintFrameworkTests/TestHelpers.swift]: Ok done in c8fa4e0. Hopefully the perf gain makes the loss of readability worthwhile in time.

Comment (keith) [inline Source/SwiftLintFrameworkTests/TestHelpers.swift]: If you don't think it will we can revert it. Either way is fine with me.

**Comment by jpsim** [comment]
Thanks for reviewing @keith and @norio-nomura

**Pull request merged**